streams/src/main/java/org/apache/kafka/streams/state/KeyValueStore.java

```diff
@@ -97,4 +97,14 @@
      */
     KeyValueIterator<K, V> all();
 
+    /**
+     * Return an approximate count of key-value mappings in this store.
+     *
+     * The count is not guaranteed to be exact in order to accommodate stores
+     * where an exact count is expensive to calculate.
+     *
+     * @return an approximate count of key-value mappings in the store.
+     */
+    long approximateNumEntries();
```

Comment:
I've updated this signature from `int size()` to `long approxiateNumEntries`. Also updated javadoc.